File: go/deposit-profit/deposit-profit.go
package main

import(
	"fmt"
)

func depositProfit(deposit, rate, threshold int)int{
	n := 0
    for deposit < threshold{
        n++
        deposit += deposit * rate / 100
    }
    
    return n
}

func main(){
	deposit, rate, threshold := 100, 20, 170

	fmt.Printf("To pass the threshold %d with an initial deposit of %d and with an annual rate of %d%% you'll need %d years", threshold, deposit, rate, depositProfit(deposit, rate, threshold))
}

File: js/deposit-profit.js
function depositProfit(deposit, rate, threshold){
    let n = 0
    while(deposit < threshold){
        n++
        deposit += deposit * rate / 100
    }
    
    return n
}

let deposit = 100, rate = 20, threshold = 170

console.log(`To pass the threshold ${threshold} with an initial deposit of ${deposit} and with an annual rate of ${rate}% you'll need`, depositProfit(deposit, rate, threshold), `years`)